### PR TITLE
GEODE-1279: Rename regression tests

### PR DESCRIPTION
**Pull request by kirklund**
These are the last few IntegrationTests and DistributedTests renamed from "Bug<TRAC#>Test" to something meaningful to Geode community.

I also overhauled tests to use JUnit 4 Rules and fixed some flakiness.

### CODE REVIEW

**Comment by kirklund** [comment]
@dhemery please review this PR!

File: geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ClientDestroysRegionRegressionTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    port2 = server2.invoke(() -> createServerCache());
+
+    createClientCache();
+  }
+
+  /**
+   * This tests whether the region destroy are not received by the sender
+   */
+  @Test
+  public void testRegionDestroyNotReceivedBySender() throws Exception {
```
Comment (dhemery) [inline geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ClientDestroysRegionRegressionTest.java]: Consider: `senderDoesNotReceiveRegionDestroy()`

File: geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/HARegionQueueStatsCloseRegressionTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+   * This test does the following: <br>
+   * 1) Create the client <br>
+   * 2) Do some operations from the cache-server <br>
+   * 3) Stop the primary cache-server <br>
+   * 4) Explicitly close the CacheClientProxy on the server. <br>
+   * 5) Verify that HARegionQueue stats are closed and entry for the ha region is removed from
+   * dispatchedMessagesMap.
+   */
+  @Test
+  public void testHAStatsCleanup() throws Exception {
```
Comment (dhemery) [inline geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/HARegionQueueStatsCloseRegressionTest.java]: Can we get more of the responsibility into the test name? (I can't think of a concise way to describe the responsibility.)

File: geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/RegisterInterestOnServerWithoutRegionRegressionTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    int port = server.invoke(() -> createServerCache());
+
+    client.invoke(() -> createClientCache(port));
+
+    client.invoke(() -> awaitConnectedServerCount(1));
+  }
+
+  @Test
+  public void registerInterestDoesNotRequireRegionOnServer() {
```
Comment (dhemery) [inline geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/RegisterInterestOnServerWithoutRegionRegressionTest.java]: I like `Does` (as in this name) way better than `Should` (as in some of the other names).

Comment (kirklund) [reply geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/RegisterInterestOnServerWithoutRegionRegressionTest.java]: I updated some class test names and test method names.

**Pull request merged**